drivers/power/adm1177/adm1177.h

```diff
@@ -0,0 +1,124 @@
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+******************************************************************************/
+#ifndef __ADM1177_H__
+#define __ADM1177_H__
+
+#include <stdint.h>
+#include <stdio.h>
+#include "no_os_i2c.h"
+#include "no_os_util.h"
+
+/* ADM1177 Slave Address */
+#define	ADM1177_ADDRESS		0x5A
+
+/* ADM1177 Write Command Byte Bits */
+#define ADM1177_CMD_V_CONT      	NO_OS_BIT(0)
+#define ADM1177_CMD_V_ONCE      	NO_OS_BIT(1)
+#define ADM1177_CMD_I_CONT      	NO_OS_BIT(2)
+#define ADM1177_CMD_I_ONCE      	NO_OS_BIT(3)
+#define ADM1177_CMD_VRANGE      	NO_OS_BIT(4)
+#define ADM1177_CMD_STATUS_RD   	NO_OS_BIT(6)
+
+/* ADM11777 Write Extended Command Byte Register Addresses */
+#define ADM1177_REG_ALERT_EN    	(NO_OS_BIT(7) | 0x1)
+#define ADM1177_REG_ALERT_TH    	(NO_OS_BIT(7) | 0x2)
+#define ADM1177_REG_CONTROL     	(NO_OS_BIT(7) | 0x3)
+
+/* ALERT_EN Register Operations */
+#define ADM1177_EN_ADC_OC1_MASK 	NO_OS_BIT(0)
+#define ADM1177_EN_ADC_OC4_MASK 	NO_OS_BIT(1)
+#define ADM1177_EN_HS_ALERT_MASK     	NO_OS_BIT(2)
+#define ADM1177_EN_OFF_ALERT_MASK    	NO_OS_BIT(3)
+#define ADM1177_CLEAR_MASK           	NO_OS_BIT(4)
+
+/* CONTROL Register Operations */
+#define ADM1177_SWOFF_MASK           	NO_OS_BIT(0)
+
+/* Status Byte Operations */
+#define ADM1177_ADC_OC_MASK    		NO_OS_BIT(0)
+#define ADM1177_ADC_ALERT_MASK 		NO_OS_BIT(1)
+#define ADM1177_HS_OC_MASK     		NO_OS_BIT(2)
+#define ADM1177_HS_ALERT_MASK  		NO_OS_BIT(3)
+#define ADM1177_OFF_STATUS_MASK		NO_OS_BIT(4)
+#define ADM1177_OFF_ALERT_MASK 		NO_OS_BIT(5)
+
+#define ADM1177_V_FULLSCALE_0   26350
+#define ADM1177_V_FULLSCALE_1   6650
+#define ADM1177_R_SENSE         25
+
+enum adm1177_chan_en {
+	ADM1177_VOLTAGE_EN,
+	ADM1177_CURRENT_EN,
+	ADM1177_VOLTAGE_AND_CURRENT_EN,
+};
+
+struct adm1177_dev {
+	struct no_os_i2c_desc   *i2c_desc;
+
+	enum adm1177_chan_en chan_en;
+};
+
+struct adm1177_init_param {
+	struct no_os_i2c_init_param     i2c_init;
+};
+
+int adm1177_init(struct adm1177_dev **device,
+		 struct adm1177_init_param *init_param);
+
+int adm1177_remove(struct adm1177_dev *device);
```

Comment:
either add header comment for all function prototypes or none of them.

Reply:
Will add header comment for all function prototypes.